test/runtests.jl

```diff
@@ -61,3 +61,193 @@ cd(Pkg.dir("Coverage")) do
     @test isempty(Coverage.process_cov("fakefile",datadir))
     rm("fakefile")
 end
+
+
+
+
+######################
+# codecovio.jl tests #
+######################
+
+"""
+extracts the api URL from stdout in a codecov.io submit call
+very helpful for testing codecovio.jl
+"""
+function extract_codecov_url(fun)
+    originalSTDOUT = STDOUT
+
+    (outRead, outWrite) = redirect_stdout()
+
+    fun()
+
+    close(outWrite)
+
+    data = utf8(readavailable(outRead))
+
+    close(outRead)
+    redirect_stdout(originalSTDOUT)
+
+    lines = split(data, "\n")
+
+    url = "None"
+    get_next = false
+    for line in lines
+        if get_next 
+            url = line
+            get_next = false
+        end
+        if contains(line, "Codecov.io API URL")
+            get_next = true
+        end
+    end
+
+    #println("url: $(url)")
+    @assert url != "None" "unable to find codecov api url in stdout, check for changes in codecovio.jl" 
+    return url
+end
+
+
+
+# empty file coverage for testing
+fcs = FileCoverage[] 
+
+# setup base system ENV vars for testing
+withenv(   
+    "CODECOV_URL" => nothing,
+    "CODECOV_TOKEN" => nothing,
+    "TRAVIS_BRANCH" => nothing,
+    "TRAVIS_COMMIT" => nothing,
+    "TRAVIS_PULL_REQUEST" => nothing,
+    "TRAVIS_JOB_ID" => nothing,
+    "TRAVIS_REPO_SLUG" => nothing,
+    "TRAVIS_JOB_NUMBER" => nothing
+    ) do 
+
+    # test local submission process
+
+    # default values
+    codecov_url = extract_codecov_url( () -> Coverage.Codecov.submit_local(fcs; dry_run = true) )
+    @test contains(codecov_url, "codecov.io")
+    @test contains(codecov_url, "commit")
+    @test contains(codecov_url, "branch")
+    @test !contains(codecov_url, "service")
+
+    # default values in depreciated call
+    codecov_url = extract_codecov_url( () -> Coverage.Codecov.submit_token(fcs) )
```

Comment:
@tkelman is there a standard/simple way that tests can check for a warning?

This [thread](https://groups.google.com/forum/#!msg/julia-users/X90Tvh4kixU/DNbB-wJFacAJ) suggests there is not a simple way to do it.